build-tools/packages/build-cli/src/commands/generate/sourceEntrypoints.ts

```diff
@@ -0,0 +1,410 @@
+			// value we use in our base config, so it should be a safe value.
+			jsx: 2 /* JSXEmit.React */,
+		},
+	});
+	const mainSourceFile = project.addSourceFileAtPath(mainEntrypoint);
+	const exports = getApiExports(mainSourceFile);
+
+	const packageDocumentationHeader = getPackageDocumentationText(mainSourceFile);
+	const newFileHeader = `${generatedHeader}${packageDocumentationHeader}`;
+
+	// This order is critical as alpha should include beta should include public.
+	// Legacy is separate and should not be included in any other level. But it
+	// may include public.
+	//   (public) -> (legacy)
+	//           `-> (beta) -> (alpha)
+	const apiTagLevels: readonly Exclude<ApiTag, typeof ApiTag.internal>[] = [
+		ApiTag.public,
+		ApiTag.legacy,
+		ApiTag.beta,
+		ApiTag.alpha,
+	] as const;
+	let commonNamedExports: Omit<ExportSpecifierStructure, "kind">[] = [];
+
+	if (exports.unknown.size > 0) {
+		log.errorLog(
+			`${exports.unknown.size} export(s) found without a recognized API level tag:\n\t${[
+				...exports.unknown.entries(),
+			]
+				.map(
+					([name, { exportedDecl, exportDecl }]) =>
+						`${name} from ${sourceContext(exportedDecl)}${
+							exportDecl === undefined ? "" : ` via ${sourceContext(exportDecl)}`
+						}`,
+				)
+				.join(`\n\t`)}`,
+		);
+
+		// Export all unrecognized APIs preserving behavior of api-extractor roll-ups.
+		for (const name of [...exports.unknown.keys()].sort()) {
+			commonNamedExports.push({ name, leadingTrivia: "\n\t" });
+		}
+		commonNamedExports[0].leadingTrivia = `\n\t// Unrestricted APIs\n\t`;
+		commonNamedExports[commonNamedExports.length - 1].trailingTrivia = "\n";
+	}
+
+	for (const apiTagLevel of apiTagLevels) {
+		const namedExports = [...commonNamedExports];
+
+		// Append this level's additional (or only) exports sorted by ascending case-sensitive name
+		const orgLength = namedExports.length;
+		const levelExports = [...exports[apiTagLevel]].sort((a, b) => (a.name > b.name ? 1 : -1));
+		for (const levelExport of levelExports) {
+			namedExports.push({ ...levelExport, leadingTrivia: "\n\t" });
+		}
+		if (namedExports.length > orgLength) {
+			namedExports[orgLength].leadingTrivia = `\n\t// @${apiTagLevel} APIs\n\t`;
+			namedExports[namedExports.length - 1].trailingTrivia = "\n";
+		}
+
+		// legacy APIs do not accumulate to others
+		if (apiTagLevel !== "legacy") {
+			// Additionally, if beta should not accumulate to alpha (alpha may be
+			// treated specially such as mapped to /legacy) then skip beta too.
+			// eslint-disable-next-line unicorn/no-lonely-if
+			if (apiTagLevel !== "beta") {
+				// update common set
+				commonNamedExports = namedExports;
+			}
+		}
+
+		const output = mapApiTagLevelToOutput.get(apiTagLevel);
+		if (output === undefined) {
+			continue;
+		}
+
+		const outFile = output.relPath;
+		log.info(`\tGenerating ${outFile}`);
+		const sourceFile = project.createSourceFile(outFile, undefined, {
```

Comment:
This logic should have some lines to create the target directory if it doesn't exist.
(It is getting by since current use expects tsconfig next to the target source location.)